packages/medusa/src/services/cart.ts

```diff
@@ -900,10 +901,36 @@ class CartService extends TransactionBaseService {
           }
 
           if (currentItem) {
+            const variantsPricing = await this.pricingService_
+              .withTransaction(transactionManager)
+              .getProductVariantsPricing(
+                [
+                  {
+                    variantId: item.variant_id!,
+                    quantity: item.quantity,
+                  },
+                ],
+                {
+                  region_id: cart.region_id,
+                  customer_id: cart.customer_id,
+                  include_discount_prices: true,
+                }
+              )
+
+            const { calculated_price } =
+              variantsPricing[currentItem.variant_id!]
+
             lineItemsToUpdate[currentItem.id] = {
               quantity: item.quantity,
               has_shipping: false,
             }
+
+            if (
+              isDefined(calculated_price) &&
+              currentItem.unit_price !== calculated_price
```

Comment:
todo: I think we can assign the calculated prices even though it is the same, it shouldn't be a problem and we can remove the second condition. wdyt?

Reply:
sure, doesn't make a difference 😄